core/templates/core/widget_tag_editor.html

```diff
@@ -0,0 +1,233 @@
+        </div>
+        <span>Recently used:</span>
+        {% for t in widget.tag_suggestions.quick %}
+        <span data-value="{{ t.name }}" data-color="{{ t.color }}" class="tag btn badge badge-pill cursor-pointer mr-1" style="background-color: {{ t.color }};">
+            {{ t.name }}
+            <span class="add-remove-icon pl-1 pr-1">+</span>
+        </span>
+        {% endfor %}
+    </div>
+    <input 
+        type="hidden"
+        name="tags"
+        value="{% for t in widget.value %}&quot;{{ t.name }}&quot;{% if not forloop.last %},{% endif %}{% endfor %}"
+    >
+<script>
+        window.addEventListener('load', () => {
+            const widget = document.getElementById('{{ widget.attrs.id }}');
+            const csrfToken = document.querySelector('input[name="csrfmiddlewaretoken"]').value;
+            
+            const prototype = widget.querySelector('.prototype-tag');
+            const currentTags = widget.querySelector('.current_tags');
+            const newTags = widget.querySelector('.new-tags');
+
+            const inputElement = widget.querySelector('input[type="hidden"]');
+
+            const apiTagsUrl = widget.getAttribute('data-tags-url');
+            const createTagInputs = widget.querySelector('.create-tag-inputs');
+            const addTagInput = createTagInputs.querySelector('input[type="text"]');
+            const addTagButton = createTagInputs.querySelector('.btn-add-new-tag');
+
+            function doReq(method, uri, data, success, fail) {
+                const req = new XMLHttpRequest();
+                req.addEventListener('load', () => {
+                    if ((req.status >= 200) && (req.status < 300)) {
+                        success(req.responseText, req);
+                    } else {
+                        fail(req.responseText, req);
+                    }
+                });
+                for (const name of ["error", "timeout", "abort"]) {
+                    req.addEventListener(name, () => {
+                        fail(req.responseText, req);
+                    });
+                }
+                req.timeout = 20000;
+
+                req.open(method, uri);
+                req.setRequestHeader("Content-Type", "application/json");
+                req.setRequestHeader("Accept", "application/json");
+                req.setRequestHeader("X-CSRFTOKEN", csrfToken);
+                req.send(data);
+            }
+
+            function createTagClicked() {
+                const tagName = addTagInput.value.trim();
+                const uriTagName = encodeURIComponent(tagName);
+
+                function success() {
+                    addTagInput.value = "";
+                }
+
+                function fail(msg) {
+                    msg = msg || "Error creating tag";
+
+                    addTagInput.select()
+                    alert(msg);
+                }
+
+                if (!tagName) {
+                    fail('Not a valid tag name');
```

Comment:
This will need to be translatable. I have never looked in to string translation in JS with Django so not sure how challenging that will be. Perhaps we can just refactor this string in to the twig template.